### PR TITLE
Support for processing forward messages

### PR DESCRIPTION
**Pull request by nain-F49FF806**

### CODE REVIEW

File: agents/rust/mediator/mediation/src/didcomm_types.rs [hunk 1]
```diff
@@ -24,22 +24,22 @@ pub struct ForwardMsg {
...
-//     pub fn new(recipient_key: &str, message: &str) -> ForwardMsg {
-//         ForwardMsg {
-//             _type: type_uri::FORWARD.to_string(),
-//             recipient_key: recipient_key.to_string(),
-//             message: message.to_string(),
-//         }
-//     }
-// }
+impl ForwardMsg {
+    // pub fn default_alice() -> ForwardMsg {
```
Comment (Patrik-Stas) [inline agents/rust/mediator/mediation/src/didcomm_types.rs:28]: delete if not needed

Comment (nain-F49FF806) [reply agents/rust/mediator/mediation/src/didcomm_types.rs]: Done

**Review by Patrik-Stas — CHANGES_REQUESTED** [review]
left more comments which came to my mind as part of reviewing via DM, most of which are out of scope for this particular PR

File: agents/rust/mediator/tests/mediator-routing-forward.rs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+use crate::common::{
+    agent_and_transport_utils::{
+        gen_mediator_connected_agent, send_message_and_pop_response_message,
+    },
+    prelude::*,
+    test_setup::OneTimeInit,
+};
+
+struct TestSetupAries;
+impl OneTimeInit for TestSetupAries {
```
[the remark this answers is not in the document]
Comment (nain-F49FF806) [reply agents/rust/mediator/tests/mediator-routing-forward.rs:36]: Done. 

It was meant to be test-file specific setup code. But currently same setup is being run. 
So I have moved relevant setup into a common function.

File: agents/rust/mediator/tests/mediator-routing-forward.rs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        }
+        load_dot_env();
+        setup_logging();
+    }
+}
+
+async fn get_mediator_grant_data(
+    agent: &Agent<impl BaseWallet + 'static, impl MediatorPersistence>,
+    agent_aries_transport: &mut impl AriesTransport,
+    agent_verikey: &VeriKey,
```
[the remark this answers is not in the document]
Comment (nain-F49FF806) [reply agents/rust/mediator/tests/mediator-routing-forward.rs:53]: Noted.

Done.

**Pull request merged**